ethcore/src/client/config.rs

```diff
@@ -113,8 +113,8 @@ impl Default for ClientConfig {
 			verifier_type: VerifierType::Canon,
 			state_cache_size: 1 * mb,
 			jump_table_size: 1 * mb,
-			history: 64,
-			history_mem: 32 * mb,
+			history: 128,
+			history_mem: 48 * mb,
```

Comment:
how is it calculated exactly? why not e.g. 64mb?

Reply:
No calculation really. We know 32Mb is not enough (but it was until quite recently) so the code will attempt to prune quite often, and this is why snapshots now fail. Up until recently we probably stayed inside the 32Mb most of the time and so even with a 64 block history we'd actually have more than a 100 (enough to start a snapshot). By bumping to a little more we likely stay within the memory limit more often and don't have to attempt pruning.

Reply:
What do you think of removing `history_mem` limit entirely? This parameter seems very fragile and unintuitive. Number of blocks should be enough.

Reply:
I wonder if it's there to handle some kind of pathological block/series of blocks where the memory consumption went through the roof?

Reply:
Running a node I noticed that the `journal_size` was ~55Mb, so maybe 64mb is the better default here.

Reply:
> I wonder if it's there to handle some kind of pathological block/series of blocks where the memory consumption went through the roof?

There were some pretty pathological blocks during the Fall 2016 DDos attacks on Mainnet Ethereum. Not sure if that's relevant.